enclave/src/c_api/c_api.cc

```diff
@@ -1009,7 +1013,13 @@ XGB_DLL int XGDMatrixSliceDMatrix(DMatrixHandle handle,
 XGB_DLL int XGDMatrixFree(DMatrixHandle handle) {
   API_BEGIN();
   CHECK_HANDLE();
+#ifdef __ENCLAVE__
```

Comment:
Here and below - do we need to wrap the new code in a macro? Is there any scenario in which we won't want to use the new code?

Reply:
I did this for the sake of consistency -- we do this in the rest of the code as well, so as to easily identify the portions that changed compared to the original XGBoost code. But you're right, we should consider removing these macros from the codebase, we don't need them anymore.